### PR TITLE
Add --memory flag for default Slurm scheduler

### PR DESCRIPTION
**Pull request by jdice**
## Description
- Modified default Slurm scheduler to allow memory flag
- Removed memory flag from Xsede
- Edited argparse to allow redefinition of memory default for Great Lakes

## Motivation and Context
Setting memory requirements for job submission was not possible with default Slurm scheduler. 

## Types of Changes

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:

- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: flow/project.py [hunk 1]
```diff
@@ -3590,6 +3590,7 @@ class MyProject(FlowProject):
         parser_submit = subparsers.add_parser(
             'submit',
             parents=[base_parser],
+            conflict_handler='resolve',
```
Comment (bdice) [inline flow/project.py]: The default Slurm environment now defines a `--memory` option with no default value, and the Great Lakes environment defines a `--memory` option with a default of `4g`. To allow the Great Lakes option to override the default option, we needed to use `conflict_handler='resolve'`. I tested specifying and not specifying a `--memory` option for a default Slurm cluster and on Great Lakes, and it respected the default (or lack of default) properly.

**Comment by bdice** [comment]
Currently the meaning of the `--memory` flag is inconsistent across schedulers. Some clusters (Great Lakes) require that the value includes a suffix like `g` to indicate that it's in gigabytes, while other clusters (Bridges, Comet) assume that the input is a number and append `g` to it in the template. I would like to unify these by requiring the user to provide a suffix and offering an example. I am going to push that commit to a new branch because it changes the behavior on existing clusters.

**Comment by bdice** [comment]
@mikemhenry Can you review? Thanks!

**Comment by mikemhenry** [comment]
Can you fix the issues with `contributors.yaml ` that just happened from the last PR I merged in?

**Pull request merged**